lib/routes/receive_payment/lnurl/lnurl_withdraw_page.dart

```diff
@@ -0,0 +1,344 @@
+                color: themeData.primaryColor.withOpacity(0.5),
+              ),
+            );
+          }
+
+          _lightningLimits = snapshot.lightningPaymentLimits!;
+
+          final minWithdrawable = max(
+            _lightningLimits.receive.minSat.toInt(),
+            widget.requestData.minWithdrawable.toInt() ~/ 1000,
+          );
+          isBelowPaymentLimit = widget.requestData.maxWithdrawable.toInt() ~/ 1000 < minWithdrawable;
+
+          if (isBelowPaymentLimit) {
+            return BlocBuilder<CurrencyCubit, CurrencyState>(
+              builder: (context, currencyState) {
+                return Center(
+                  child: Text(
+                    texts.invoice_payment_validator_error_payment_below_invoice_limit(
+                      currencyState.bitcoinCurrency.format(minWithdrawable),
+                    ),
+                    textAlign: TextAlign.center,
+                  ),
+                );
+              },
+            );
+          }
+
+          final maxWithdrawable = min(
+            _lightningLimits.receive.maxSat.toInt(),
+            widget.requestData.maxWithdrawable.toInt() ~/ 1000,
+          );
+          return Padding(
+            padding: const EdgeInsets.only(bottom: 40.0),
+            child: SingleChildScrollView(
+              // TODO: Extract these into widgets
+              child: receivePaymentResponse == null
+                  ? _buildForm(
+                      minWithdrawable: minWithdrawable,
+                      maxWithdrawable: maxWithdrawable,
+                    )
+                  : _buildQRCode(),
+            ),
+          );
+        },
+      ),
+      bottomNavigationBar: (receivePaymentResponse == null && !isBelowPaymentLimit)
+          ? SingleButtonBottomBar(
+              stickToBottom: true,
+              text: texts.invoice_action_redeem,
+              onPressed: () {
+                if (_formKey.currentState?.validate() ?? false) {
+                  _withdraw();
+                }
+              },
+            )
+          : SingleButtonBottomBar(
+              stickToBottom: true,
+              text: texts.qr_code_dialog_action_close,
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+            ),
+    );
+  }
+
+  Widget _buildForm({
+    required int minWithdrawable,
+    required int maxWithdrawable,
+  }) {
+    final texts = context.texts();
+    final data = widget.requestData;
+
+    return BlocBuilder<CurrencyCubit, CurrencyState>(
+      builder: (context, currencyState) {
+        final isFixedAmount = data.minWithdrawable == data.maxWithdrawable;
+        final minWithdrawableFormatted = currencyState.bitcoinCurrency.format(minWithdrawable);
+        final maxWithdrawableFormatted = currencyState.bitcoinCurrency.format(maxWithdrawable);
+
+        final isFixedAmountWithinLimits = minWithdrawable == maxWithdrawable;
+        if (!isFixedAmountWithinLimits) {
```

Comment:
Is this right? Seems inconsistent with the error text and the min/max is already checked

Reply:
Explained in Slack and [added a different message if fixed payment amount is outside allowed limits](https://github.com/breez/misty-breez/pull/167/commits/ab87441912a0365c837c9e250447fd3db6ad5b35)